tests/MoBi.Tests/Presentation/Tasks/SimulationRunnerSpecs.cs

```diff
@@ -315,4 +320,165 @@ public void should_update_the_molecule_weight_of_all_concentration_columns_relat
+            runTask.Wait();
+         }
+         catch (AggregateException ex) when (ex.InnerException is TaskCanceledException)
+         {
+            // This is expected since we cancelled the task, no action needed
+            //only for the test not to fail
+         }
+      }
+
+      [Observation]
+      public void should_have_cancelled_the_simulation()
+      {
+         wasCancelled.ShouldBeTrue();
+      }
+
+      [Observation]
+      public void should_publish_simulation_canceled_event()
+      {
+         A.CallTo(() => _context.PublishEvent(A<SimulationRunCanceledEvent>._)).MustHaveHappened();
+      }
+   }
+
+   public class When_stopping_all_simulations : concern_for_SimulationRunner
+   {
+      private IMoBiSimulation _simulation1;
+      private IMoBiSimulation _simulation2;
+      private ISimModelManager _simModelManager;
+      private CancellationToken token1;
+      private CancellationToken token2;
+      private bool wasCancelled1 = false;
+      private bool wasCancelled2 = false;
+
+      //This test might fail when debugging if more time is taken than the await Task.Delay(10000, tokenx);
+      //So if debug is needed, maybe set it longer before debugging.
+
+      protected override void Context()
+      {
+         base.Context();
+
+         _simulation1 = A.Fake<IMoBiSimulation>();
+         _simulation2 = A.Fake<IMoBiSimulation>();
+         _simModelManager = A.Fake<ISimModelManager>();
+
+         A.CallTo(() => _simModelManagerFactory.Create()).Returns(_simModelManager);
+
+         A.CallTo(() => _simModelManager.RunSimulationAsync(_simulation1, A<CancellationToken>._, null))
+            .Invokes(call => token1 = call.GetArgument<CancellationToken>(1))
+            .ReturnsLazily(async () =>
+            {
+               try
+               {
+                  await Task.Delay(10000, token1);
+               }
+               catch (OperationCanceledException)
+               {
+                  wasCancelled1 = true;
+                  throw;
+               }
+
+               return new SimulationRunResults(warnings: Enumerable.Empty<SolverWarning>(), results: new DataRepository("R1"));
+            });
+
+         A.CallTo(() => _simModelManager.RunSimulationAsync(_simulation2, A<CancellationToken>._, null))
+            .Invokes(call => token2 = call.GetArgument<CancellationToken>(1))
+            .ReturnsLazily(async () =>
+            {
+               try
+               {
+                  await Task.Delay(10000, token2);
+               }
+               catch (OperationCanceledException)
+               {
+                  wasCancelled2 = true;
+                  throw;
+               }
+
+               return new SimulationRunResults(warnings: Enumerable.Empty<SolverWarning>(), results: new DataRepository("R2"));
+            });
+
+         A.CallTo(() => _eventValidationTask.Validate(A<IMoBiSimulation>._)).Returns(true);
+      }
+
+      protected override void Because()
```

Comment:
Shouldn't this whole test be using the Async base instead of running the test with Wait all the time? 
It's just a quesiton because it's very hard to follow